pinot-tools/src/main/java/com/linkedin/pinot/tools/backfill/BackfillSegmentUtils.java

```diff
@@ -168,17 +163,16 @@ public boolean uploadSegment(String segmentName, File segmentDir, File outputDir
     try {
       TarGzCompressionUtils.createTarGzOfDirectory(segmentDir.getAbsolutePath(), segmentTar.getAbsolutePath());
       LOGGER.info("Created tar of {} at {}", segmentDir.getAbsolutePath(), segmentTar.getAbsolutePath());
-      InputStream inputStream = new FileInputStream(segmentTar);
-      int status = FileUploadUtils.sendSegmentFile(_controllerHost, _controllerPort, segmentName, inputStream, segmentTar.length());
-      if (status != 200) {
+      int statusCode =
+          FileUploadUtils.uploadSegment(_controllerHost, Integer.parseInt(_controllerPort), segmentName, segmentTar);
+      if (statusCode != HttpStatus.SC_OK) {
```

Comment:
how do we distinguish between transient and non-transient exceptions?

Reply:
Client decides that based on the status code, and do retry (you can use the RetryPolicy)